plugins/pipelines/badge.js

```diff
@@ -104,38 +104,42 @@ module.exports = () => ({
                         return reply.redirect(getUrl(badgeService));
                     }
 
-                    return pipeline.getEvents({ sort: 'ascending' }).then((events) => {
-                        const lastEvent = events.pop();
+                    return pipeline.getEvents({ sort: 'ascending' }).then((allEvents) => {
+                        const getLastEffectiveEvent = async (events) => {
```

Comment:
There is no `await` in this function, so this `async` shouldn't be needed.

Reply:
fixed